context/src/env/validator.rs

```diff
@@ -248,12 +248,12 @@ pub fn validate(ci_info: &CIInfo) -> EnvValidation {
                     EnvValidationIssueInvalid::CIInfoPRNumberMissing,
                 ));
             }
-            (BranchClass::Merge | BranchClass::ProtectedBranch, Some(..)) => {
+            (BranchClass::ProtectedBranch, Some(..)) => {
                 env_validation.add_issue(EnvValidationIssue::Invalid(
                     EnvValidationIssueInvalid::CIInfoPRNumberConflictsWithBranchClass,
                 ));
             }
-            (BranchClass::PullRequest, Some(..))
+            (BranchClass::PullRequest | BranchClass::Merge, Some(..))
```

Comment:
It seems like there is some inconsistency with how merge branches are implemented. We don't set the PR for Trunk merge, but Gitlab does appear to set the PR number in the CI environment for merge trains

Reply:
Yep, you are correct. I think we can ignore the PR number for GitLab merge trains 👍